reader.go

```diff
@@ -41,31 +41,20 @@ func (d *Reader) Scan() bool {
 			}
 			continue
 		}
-		if !strings.HasPrefix(line, ": ") {
-			d.err = fmt.Errorf("invalid histfile line: %q", line)
-			return false
-		}
-		i := strings.IndexRune(line[2:], ':')
-		if i < 0 {
-			d.err = fmt.Errorf("invalid histfile line: %q", line)
-			return false
-		}
-		time, err = strconv.Atoi(line[2 : i+2])
-		if err != nil {
-			d.err = fmt.Errorf("invalid histfile line: %q", line)
-			return false
-		}
-		j := strings.IndexRune(line[2:], ';')
-		if j < 0 {
-			d.err = fmt.Errorf("invalid histfile line: %q", line)
-			return false
+
+		switch {
+		case strings.HasPrefix(line, ": "):
```

Comment:
Prefer if statement, sorry for nit pick.

Reply:
Are you ok with this style?
```go
cmd = line
if strings.HasPrefix(line, ": ") {
    time, elapsed, cmd, err = extended(line)
}
```

I generally do everything I can to stay away from `else`.

Reply:
Use else statement, that's fine.